browser/net/brave_request_handler.cc

```diff
@@ -173,10 +173,7 @@ int BraveRequestHandler::OnHeadersReceived(
   ctx->override_response_headers = override_response_headers;
   ctx->allowed_unsafe_redirect_url = allowed_unsafe_redirect_url;
 
-  base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
-                           base::Bind(&BraveRequestHandler::RunNextCallback,
-                                      weak_factory_.GetWeakPtr(),
-                                      ctx));
+  RunNextCallback(ctx);
```

Comment:
Previously the async call was needed to keep proper callbacks flow in `url_request_http_job.cc` (the response callback was called synchronously and this could break things). Now response callbacks are called asynchronously (see RunCallbackForRequestIdentifier) so we can keep normal flow here.